### PR TITLE
ddl: fix show global index regions

### LINKED ISSUE #46000: show table regions is wrong for table with global index
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

a. enable-global-index=true in config file.
b. create table and execute show ddl.
```SQL
create table p (id int, c int, d int, unique key uidx(c), unique key idx(id)) partition by range (c) (
partition p0 values less than (4),
partition p1 values less than (7),
partition p2 values less than (10))
```
```SQL
show table p regions
```

### 2. What did you expect to see? (Required)
image

### 3. What did you see instead (Required)
image

### 4. What is your TiDB version? (Required)
image

### PR DESCRIPTION
**Pull request by L-maple**
### What problem does this PR solve?

Issue Number: close #46000

Problem Summary:
`show table t regions` is not right for table with global index. This PR is to fix it.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

File: executor/show.go [hunk 1]
```diff
@@ -2039,9 +2047,16 @@ func (e *ShowExec) fetchShowTableRegions(ctx context.Context) error {
...
-		regions, err = getTableIndexRegions(indexInfo, physicalIDs, tikvStore, splitStore)
+		if indexInfo.Global {
+			regions, err = getTableIndexRegions(indexInfo, []int64{tb.Meta().ID}, tikvStore, splitStore)
+		} else {
+			regions, err = getTableIndexRegions(indexInfo, physicalIDs, tikvStore, splitStore)
+		}
 	} else {
 		// show table * region
+		if hasGlobalIndex {
+			physicalIDs = append([]int64{tb.Meta().ID}, physicalIDs...)
```
Comment (mjonss) [inline executor/show.go:2058]: I wonder if this is more efficient and easier to read?
```suggestion
			physicalIDs = append(physicalIDs, tb.Meta().ID)
```

Comment (L-maple) [reply executor/show.go:2058]: @mjonss Thanks for your review~
This slice element sequence affects the display of regions. 
I think both styles are OK, maybe setting logical table in front of the partitioned table is nicer for readers. If you think the later efficient one is better, I will propose a new PR.

Comment (mjonss) [reply executor/show.go:2058]: Ah, that is an OK reason, to have the table level regions first.

**Pull request merged**